conda_build/build_index.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (C) 2014 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+conda-build's use of conda-index, delegated to now-separate conda-index package.
+"""
+
+import conda_index.index
+
+import json
+import os
+from os.path import (
+    dirname,
+)
+import sys
+import time
+
+from functools import partial
+import logging
+
+from concurrent.futures import Executor
+
+from conda_build import conda_interface, utils
+from .conda_interface import context
+from .conda_interface import CondaHTTPError, get_index, url_path
+from .utils import (
+    JSONDecodeError,
+    get_logger,
+)
+
+log = get_logger(__name__)
+
+
+# use this for debugging, because ProcessPoolExecutor isn't pdb/ipdb friendly
+class DummyExecutor(Executor):
+    def map(self, func, *iterables):
+        for iterable in iterables:
+            for thing in iterable:
+                yield func(thing)
+
+
+try:
+    from conda.base.constants import NAMESPACES_MAP, NAMESPACE_PACKAGE_NAMES
+except ImportError:
+    NAMESPACES_MAP = {  # base package name, namespace
+        "python": "python",
+        "r": "r",
+        "r-base": "r",
+        "mro-base": "r",
+        "mro-base_impl": "r",
+        "erlang": "erlang",
+        "java": "java",
+        "openjdk": "java",
+        "julia": "julia",
+        "latex": "latex",
+        "lua": "lua",
+        "nodejs": "js",
+        "perl": "perl",
+        "php": "php",
+        "ruby": "ruby",
+        "m2-base": "m2",
+        "msys2-conda-epoch": "m2w64",
+    }
+    NAMESPACE_PACKAGE_NAMES = frozenset(NAMESPACES_MAP)
+    NAMESPACES = frozenset(NAMESPACES_MAP.values())
```

Comment:
Is this used?

Reply:
it wouldn't be surprising if conda.base.constants exists in all conda > 4.7 or the oldest one we can depend on.